remix-solidity/src/compiler/compiler-input.ts

```diff
@@ -25,9 +27,9 @@ module.exports = (sources, opts) => {
     o.language = opts.language
   }
   if (opts.language === 'Yul' && o.settings.optimizer.enabled) {
-    // details key is valid only for Yul
-    if (!o.settings.optimizer.details) o.settings.optimizer.details = {}
-    o.settings.optimizer.details['yul'] = true
+    if (!o.settings.optimizer['details']) 
```

Comment:
It's difficult to read. If I understand well the if statement apply on the `o.settings.optimizer['details'] = {}`. If this is the case better use {} (even if it add one line).

Reply:
This is checking if `o.settings.optimizer['details']` is null or undefined and in that case it initialise `o.settings.optimizer['details'] = {}` so `o.settings.optimizer['details']['yul']`  doesn't throw the error

Reply:
Ok, for better readibility can you do : 
```typescript
if (!o.settings.optimizer['details']) {
  o.settings.optimizer['details'] = {}
}
```